src/systems/variational_smoother_constraint.C

```diff
@@ -521,6 +521,47 @@ void VariationalSmootherConstraint::constrain_node_to_line(const Node & node,
     }
 }
 
+void VariationalSmootherConstraint::find_nodal_or_face_neighbors(
+    const MeshBase &mesh, const Node &node,
+    const std::unordered_map<dof_id_type, std::vector<const Elem *>>
+        &nodes_to_elem_map,
+    std::vector<const Node *> &neighbors)
+{
+  // Find all the nodal neighbors... that is the nodes directly connected
+  // to this node through one edge.
+  MeshTools::find_nodal_neighbors(mesh, node, nodes_to_elem_map, neighbors);
+
+  // If no neighbors are found, use all nodes on the containing side as
+  // neighbors.
+  if (!neighbors.size())
+    {
+      // Grab the element containing node
+      const auto *elem = nodes_to_elem_map.at(node.id()).front();
```

Comment:
Always use `libmesh_map_find()` instead of `map::at()`. The former gives a nicely formatted error message in the exception when the key is not found, the latter just throws a generic std::exception.